opensearch-project/OpenSearch-Dashboards
# [Workspace]Add flag for workspace collaborator editor enabled

### Description

This PR add a feature flag to determine whether collaborator editor showed in workspace creation and workspace detail page. Beside this, this PR add two methods to register workspace creation post processor and register customized tab in workspace details page.

## Screenshot


### Without collaborator editor enabled
![image](https://github.com/user-attachments/assets/1a40e62b-84ed-498c-9377-1b183ff038f5)
![image](https://github.com/user-attachments/assets/5bddd5b1-f60a-41fd-9aeb-38feb249cebf)

### With collaborator editor enabled
![image](https://github.com/user-attachments/assets/dfd9537d-cd4a-41e6-b393-be29b9bba02a)
![image](https://github.com/user-attachments/assets/d8697041-f973-4bdb-9704-cfe7bb18642c)


## Testing the changes



- Clone branch code and run `yarn osd bootstrap --single-version loose`
- Add configs below in `config/opensearch_dashboards.yml`
```
savedObjects.permission.enabled: true
workspace.enabled: true
uiSettings:
  overrides:
    'home:useNewHomePage': true
opensearchDashboards.dashboardAdmin.users: ['admin']
workspace.collaboratorEditorEnabled: true
```
- Run `yarn start --no-base-path`
- Login with admin user and go to workspace creation page by bottom left menu
- The collaborator editor should be exists
- Click create button, it will redirect to workspace use case landing page
- Set `workspace.collaboratorEditorEnabled` to `false` in `config/opensearch_dashboards.yml`
- The collaborator editor should be hidden

For workspace details customized tabs tests, can refer "should render customized tab" in `src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx`.
For register workspace create post processor tests, can refer "#setup able to registerWorkspaceCreationPostProcessor" in `src/plugins/workspace/public/plugin.test.ts`
 

## Changelog

- feat: [Workspace]Add flag for workspace collaborator editor enabled

### Check List
